src/main/attached_database.cpp

```diff
@@ -29,35 +106,47 @@ AttachedDatabase::AttachedDatabase(DatabaseInstance &db, AttachedDatabaseType ty
 }
 
 AttachedDatabase::AttachedDatabase(DatabaseInstance &db, Catalog &catalog_p, string name_p, string file_path_p,
-                                   AccessMode access_mode)
+                                   const AttachOptions &options)
     : CatalogEntry(CatalogType::DATABASE_ENTRY, catalog_p, std::move(name_p)), db(db), parent_catalog(&catalog_p) {
-	type = access_mode == AccessMode::READ_ONLY ? AttachedDatabaseType::READ_ONLY_DATABASE
-	                                            : AttachedDatabaseType::READ_WRITE_DATABASE;
+
+	if (options.access_mode == AccessMode::READ_ONLY) {
+		type = AttachedDatabaseType::READ_ONLY_DATABASE;
+	} else {
+		type = AttachedDatabaseType::READ_WRITE_DATABASE;
+	}
+
+	// We create the storage after the catalog to guarantee we allow extensions to instantiate the DuckCatalog.
 	catalog = make_uniq<DuckCatalog>(*this);
-	// do this after catalog to guarnatee we allow extension to instantionate DuckCatalog causing creation
-	// of the storage
-	storage = make_uniq<SingleFileStorageManager>(*this, std::move(file_path_p), access_mode == AccessMode::READ_ONLY);
+	auto read_only = options.access_mode == AccessMode::READ_ONLY;
+	storage = make_uniq<SingleFileStorageManager>(*this, std::move(file_path_p), read_only, options.block_alloc_size);
```

Comment:
Here we also need the `AttachInfo`

Or extend `AttachOptions` to somehow have a reference to all of the unused options